src/main/java/io/strimzi/kafka/bridge/http/HttpSourceBridgeEndpoint.java

```diff
@@ -158,8 +121,7 @@ public Iterator<Map.Entry<String, String>> iterator() {
         if (isAsync) {
             // if async is specified, return immediately once records are sent
             this.sendAsyncRecords(records);
-            Tags.HTTP_STATUS.set(span, HttpResponseStatus.NO_CONTENT.code());
-            span.finish();
+            span.finish(HttpResponseStatus.NO_CONTENT.code());
             HttpUtils.sendResponse(routingContext, HttpResponseStatus.NO_CONTENT.code(),
```

Comment:
Why do we finish the span before sending the response here, but the other way around in the code above? I'd have expected the order should be consistent.

Reply:
Afaik, I didn't change the order ... but will check again.
But yeah, consistent order makes sense.
Which should it be -- finish span 1st and then response, or vice-versa?

Reply:
I don't know for certain, but I would finish the span first and send the response. That way the tracing should definitely be present to clients that have observed the HTTP response.